pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -0,0 +1,102 @@
+# Gossipsub v1.2
+
+# Overview
+
+This document aims to provide a minimal extension to the [gossipsub
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to enhance the
+efficiency (minimize amplification/duplicates and decrease message latency) of
+the gossip mesh networks for larger messages. 
+
+In more specific terms, a new control message is introduced: `DONTSEND`. It's primarily 
+intended to notify mesh peers that the node already received a message and there is no 
+need to send its duplicate.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
+
+## Parameters
+
+This section lists the configuration parameters that needs to agreed on across clients to avoid 
+ peer penalizations
+
+| Parameter               | Description                                                      | Reasonable Default |
+|-------------------------|------------------------------------------------------------------|--------------|
+| `max_dontsend_messages` | The maximum number of `DONTSEND` messages per heartbeat per peer | ???  |
```

Comment:
Is that necessary? We can just descore peer if they don't send duplicate, or we don't eventually see the message id
GossipSub already has too many parameters

Reply:
> or we don't eventually see the message id

Sounds like a good idea 👍

Reply:
> or we don't eventually see the message id

Considering `DONTSEND` is allowed to be sent before validation, a peer can be downscored if a message `DONTSEND` has been sent for appears to be invalid

Reply:
Do we have numbers on how much we can gain by allowing to send `DONTSEND`s before validation? (ie, how long the validation is in practice)

Reply:
Note that validation is some networks can be slow, so there is real benefit by sending early.